lib/Service/IonosMailerService.php

```diff
@@ -7,20 +7,76 @@
 
 namespace OCA\IonosProcesses\Service;
 
+use Exception;
+use IONOS\MailNotificationAPI\Client\Model\ShareMessageByLink;
+use OCA\IonosProcesses\AppInfo\Application;
+use OCA\IonosProcesses\Listener\ShareCreatedEventListener;
+use OCP\Exceptions\AppConfigException;
+use OCP\IAppConfig;
+use Psr\Log\LoggerInterface;
+
 /**
  * Client for the internal mail delivery web service.
  */
 class IonosMailerService {
-	public function __construct() {
+	public const BRAND = 'IONOS';
+
+	public function __construct(
+		private IAppConfig $appConfig,
+		private LoggerInterface $logger,
+		private ApiClientService $apiClientService,
+	) {
 	}
 
 	/**
 	 * Send mails to the mail delivery web service
 	 *
 	 * @param string $eventName name of the event
 	 * @param array $variables variables to send to the mail service
+	 * @throws Exception
 	 */
 	public function send(string $eventName, array $variables): void {
-		// Stub
+		$apiBaseUrl = $this->appConfig->getValueString(Application::APP_ID, 'ionos_mail_base_url');
+		$allowInsecure = !$this->appConfig->getValueBool(Application::APP_ID, 'allow_insecure');
```

Comment:
Please move the negation down where `$allowInsecure` is referenced for `'verify' =>`, since, at this location this looks wrong:

```diff
index faa7257..8d7c9a2 100644
--- lib/Service/IonosMailerService.php
+++ lib/Service/IonosMailerService.php
@@ -37,7 +37,7 @@ class IonosMailerService {
         */
        public function send(string $eventName, array $variables): void {
                $apiBaseUrl = $this->appConfig->getValueString(Application::APP_ID, 'ionos_mail_base_url');
-               $allowInsecure = !$this->appConfig->getValueBool(Application::APP_ID, 'allow_insecure');
+               $allowInsecure = $this->appConfig->getValueBool(Application::APP_ID, 'allow_insecure');
                $basicAuthUser = $this->appConfig->getValueString(Application::APP_ID, 'basic_auth_user');
                $basicAuthPass = $this->appConfig->getValueString(Application::APP_ID, 'basic_auth_pass');
 
@@ -64,7 +64,7 @@ class IonosMailerService {
 
                $client = $this->apiClientService->newClient([
                        'auth' => [$basicAuthUser, $basicAuthPass],
-                       'verify' => $allowInsecure,
+                       'verify' => !$allowInsecure,
                ]);
 
                $apiInstance = $this->apiClientService->newEventAPIApi($client, $apiBaseUrl);

```